yt_dlp/extractor/ant1newsgr.py

```diff
@@ -0,0 +1,184 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+from urllib.parse import (
+    parse_qs,
+    urlparse,
+    urlunparse,
+)
+
+from .common import InfoExtractor
+from ..utils import (
+    HEADRequest,
+    ExtractorError,
+    determine_ext,
+    smuggle_url,
+    unsmuggle_url,
+    unescapeHTML,
+)
+
+
+class Ant1NewsGrBaseIE(InfoExtractor):
+    @staticmethod
+    def _smuggle_parent_info(url, **info_dict):
+        return smuggle_url(url, {'parent_info': info_dict})
+
+    @staticmethod
+    def _unsmuggle_parent_info(url):
+        unsmuggled_url, data = unsmuggle_url(url, default={'parent_info': {}})
+        return unsmuggled_url, data['parent_info']
+
+    def _download_api_data(self, netloc, cid, scheme='https'):
+        url_parts = (scheme, netloc, self._API_PATH, None, None, None)
+        url = urlunparse(url_parts)
+        query = {'cid': cid}
+        return self._download_json(
+            url, cid,
+            'Downloading JSON',
+            'Unable to download JSON',
+            query=query)
+
+    def _download_and_extract_api_data(self, video_id, *args, **kwargs):
+        info = self._download_api_data(*args, **kwargs)
+        try:
+            source = info['url']
+        except KeyError:
+            raise ExtractorError('no source found for %s' % video_id)
+        formats, subs = self._extract_m3u8_formats_and_subtitles(
+            source, video_id, 'mp4') \
+            if determine_ext(source) == 'm3u8' else ([source], {})
+        self._sort_formats(formats)
+        return {
+            'id': video_id,
+            'title': info['title'],
+            'thumbnail': info['thumb'],
+            'formats': formats,
+            'subtitles': subs,
+        }
+
+
+class Ant1NewsGrWatchIE(Ant1NewsGrBaseIE):
+    IE_NAME = 'ant1newsgr:watch'
+    IE_DESC = 'ant1news.gr videos'
+    _VALID_URL = r'https?://(?:www\.)?ant1news\.gr/watch/(?P<id>\d+)/'
+    _API_PATH = '/templates/data/player'
+
+    _TEST = {
+        'url': 'https://www.ant1news.gr/watch/1506168/ant1-news-09112021-stis-18-45',
+        'md5': '95925e6b32106754235f2417e0d2dfab',
+        'info_dict': {
+            'id': '1506168',
+            'ext': 'mp4',
+            'title': 'md5:0ad00fa66ecf8aa233d26ab0dba7514a',
+            'description': 'md5:18665af715a6dcfeac1d6153a44f16b0',
+        },
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        scheme, netloc, _, _, _, _ = urlparse(url)
```

Comment:
I don't understand why URL has to be parsed separately. Just make the netloc a named group in regex. Or just hardcode it in the API path since the extractor is only dealing with a single domain

Reply:
Both `scheme` and `netloc` could be matched in the URL. However `urlparse` is more robust in general (not in this case) and I did this also for consistency with other code that prepares parameters for calling `_download_and_extract_api_data`. I can change and avoid it.